### PR TITLE
Simplify dice even further

### PR DESCRIPTION
**Pull request by grimreaper**
This eliminates a level of abstraction that is no longer needed

### CODE REVIEW

**Comment by thpr** [comment]
This is dramatic enough, and if you are intending to use it strategically, it should be checked in with unit tests for this subsystem.

File: code/src/java/gmgen/plugin/dice/NSidedModifiedDieConfig.java [hunk 1]
```diff
@@ -20,34 +20,25 @@
...
-		};
 	}
 
 	@Override
 	public int roll()
 	{
-		return counter.totalCount(
-				ResultModifier.modify(modifiers)
-		);
+		return IntStream.generate(() -> Die.rand.nextInt(sides) + 1).map(v -> v + bias).limit(n).sum();
```
Comment (thpr) [inline code/src/java/gmgen/plugin/dice/NSidedModifiedDieConfig.java]: Format to separate rows, please

**Comment by grimreaper** [comment]
This primarily /undoes/ a leve of abstraction I introduced. Agred about unit tests. I'll add some.

File: build.gradle [hunk 2]
```diff
@@ -165,6 +165,8 @@ dependencies {
 
     testCompile group: 'junit', name: 'junit', version:'4.12'
     testCompile group: 'xmlunit', name: 'xmlunit', version:'1.3'
+    testCompile group: 'org.hamcrest', name: 'hamcrest-all', version: '1.3'
```
Comment (thpr) [inline build.gradle]: I have to admit, I'm not a fan of things just being added to build.gradle without a discussion.  I think we need more broad buy-in that things will be widely used and are value add to the project before they are just adopted

**Comment by grimreaper** [comment]
this test error confuses me, especially since it doesn't seem to be flaky

File: code/src/java/gmgen/plugin/dice/NSidedModifiedDieConfig.java (hunk 1 above)
Comment (grimreaper) [inline code/src/java/gmgen/plugin/dice/NSidedModifiedDieConfig.java]: done

Comment (thpr) [inline code/src/java/gmgen/plugin/dice/NSidedModifiedDieConfig.java]: Still seeing this as one line...

**Pull request merged**